core/commands/storage/session.go

```diff
@@ -0,0 +1,75 @@
+package storage
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/TRON-US/go-btfs/core"
+
+	"github.com/google/uuid"
+	"github.com/ipfs/go-cid"
+	coreiface "github.com/ipfs/interface-go-ipfs-core"
+)
+
+var SessionMap map[string]*Session
+
+type Session struct {
+	sync.Mutex
+
+	Time      time.Time
+	FileHash  string
+	Status    string
+	Chunks    map[string]interface{}
+	Challenge map[string]*StorageChallenge
```

Comment:
You will eventually merge `Chunks` and `Challenge` since they share the same key...

Reply:
I would build an another struct to contain all the chunks info and challenge with it. But for now, I would rather leave it there...

Reply:
ok